plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php

```diff
@@ -42,10 +42,20 @@ class WC_Email_Customer_POS_Refunded_Order extends WC_Email {
 		 */
 		public $partial_refund;
 
+		/**
+		 * Emails instance.
+		 *
+		 * @var WC_Emails
+		 */
+		private $emails;
+
 		/**
 		 * Constructor.
+		 *
+		 * @param WC_Emails|null $emails The WC_Emails instance. Used for unit testing.
```

Comment:
Not sure about this one: what exactly in the testing context does this injection?

Reply:
As the email header customization involves removing the filter from the global `WC_Emails` instance, I thought it's safer to only call it when needed. In the test suite, `WC_Emails::instance()` can be set to an instance that doesn't trigger `__construct` which sets up actions and filters (running the test class by itself passed but not when running the whole test suite, I didn't spend the time to isolate what could have affected the `WC_Emails::instance()`). Injecting the `WC_Emails` instance in tests ensures that the actions and filters are set up as expected through the constructor, and that the PR overrides the filter properly so that it doesn't result in two email headers.

Reply:
I see, this is an over-complication IMO:  `plugins/woocommerce/templates/emails/customer-pos-*.php` have `do_action( 'woocommerce_email_header', $email_heading, $email )`, which can be a different action name for the POS emails. IDK, like `woocommerce_email_pos_header` or something more meaningful.

This way, the injection can be dropped and the complexity lowered accordingly.

Reply:
Thanks for the idea! I didn't think of using a different action name before. Updated in https://github.com/woocommerce/woocommerce/pull/58124/commits/d9ad958f341adff114f18ab9898da1c8f0e29b64.

Also did another round of testing.

Reply:
Nice!